origin/src/raw_mutex.rs

```diff
@@ -8,27 +12,16 @@ use std::sync::atomic::Ordering::SeqCst;
 /// The current implementation does not provide fairness, is not reentrant,
 /// and is not proven correct.
 ///
-/// This type is not movable when locked; the `new` function is `unsafe` to
-/// reflect this.
+/// This type is not movable when locked.
```

Comment:
This is an unfortunate invariant. Can we either add more comments here explaining why we might believe that no one will accidentally try to move an instance of this type, or change `RawMutex` to have a `Box<AtomicU32>` to make it safe to move?

Reply:
using box would defeat the purpose of this lock.
added a comment.